src/unix_mode.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Stephanie Aelmore.
+// Conserve backup system.
+// Copyright 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022 Martin Pool.
+
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Discretionary Access Control permissions for archived files.
+//!
+//! On unix systems, the mode has 9 significant bits, divided into three classes,
+//! owner class, group class, and others class. The owner class permissions
+//! apply only to the owner, the group class permissions apply to members of the user's group,
+//! and the others class permissions apply to all other users.
+//!
+//! For each class, there are 3 bitflags: read, write, and execute. This is typically
+//! written as an octal number, such as 0o664, which means the user and group can
+//! both read and write, and other users can only read.
+//!
+//! The mode is also often presented as a string of characters, such as "rw-rw-r--",
+//! where each character represents one bit.
+//!
+//! On windows systems, files can be either read-only or writeable. For cross-compatibility,
+//! the mode is always stored using the unix format, where the read-only state is stored
+//! using the write bit in the user class.
+//! TODO: Properly implement and test Windows compatibility.
+//!
+use serde::{Deserialize, Serialize};
+use std::{fmt, fs::Permissions};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct UnixMode {
+    pub mode: u32,
+}
+impl Default for UnixMode {
+    fn default() -> Self {
+        // created with execute permission so that restoring old archives works properly
+        // TODO: ideally we would set this based on the inode type read from the archive
+        Self { mode: 0o100775 }
```

Comment:
I think we should probably strip off the top file-type bits and store only the permissions?

Reply:
Agreed. IIRC I had those bits in there to get the old archive test to pass, since for the most part the mode is set from reading the file metadata rather than relying on the default, but when reading archives from the testdata there were some problems.
Now that I think though it could also be accomplished with a mask in the Eq implementation rather than comparing all the bits in mode, which would be cleaner.

Reply:
Current solution is to store the inode type bits if provided but not compare them in the Eq impl, and not print them in the display impl unless they are present. I considered just stripping off the first character of the display string no matter what (as I had it previously) but I don't see how it hurts to print the inode type if the bits happen to be present.

Reply:
Regardless, it won't be hard to make further changes.

Reply:
The type bits are redundant with the separately stored file kind so I think we should strip them are the point of reading and only store the permissions in this field.

Reply:
Ok, you do have a point there.